astroid/brain/brain_pathlib.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2022 Deepyaman Datta <deepyaman.datta@utexas.edu>
+
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
+
+from typing import Iterator, Optional
+
+from astroid import bases, context, inference_tip
+from astroid.const import PY310_PLUS
+from astroid.exceptions import UseInferenceDefault
+from astroid.manager import AstroidManager
+from astroid.nodes.node_classes import Attribute, Slice, Subscript
+from astroid.nodes.node_ng import NodeNG
+
+
+def _looks_like_parents_subscript(node: NodeNG) -> bool:
+    return (
+        isinstance(node, Subscript)
+        and isinstance(node.value, Attribute)
+        and node.value.attrname == "parents"
+    )
```

Comment:
This isn't enough, it will also match unintended code. Just an example. I would suggest adding something like it as a test.
```py
name_node = astroid.extract_node(
    """\
class A:
    parents = 42

c = A()
error = c.parents[2]
error
"""
)
inferred = next(name_node.infer())
print(inferred)
```

Reply:
Done!